internal/datastore/postgres/revisions.go

```diff
@@ -250,11 +250,22 @@ func parseRevisionDecimal(revisionStr string) (datastore.Revision, error) {
 	}}, nil
 }
 
-func createNewTransaction(ctx context.Context, tx pgx.Tx) (newXID xid8, newSnapshot pgSnapshot, err error) {
+var emptyMetadata = map[string]any{}
+
+func createNewTransaction(ctx context.Context, tx pgx.Tx, metadata map[string]any) (newXID xid8, newSnapshot pgSnapshot, err error) {
 	ctx, span := tracer.Start(ctx, "createNewTransaction")
 	defer span.End()
 
-	cterr := tx.QueryRow(ctx, createTxn).Scan(&newXID, &newSnapshot)
+	if metadata == nil {
+		metadata = emptyMetadata
+	}
+
+	sql, args, err := createTxn.Values(metadata).Suffix("RETURNING xid, snapshot").ToSql()
```

Comment:
Please turn `RETURNING xid, snapshot` into a constant that is created out of `colXID` and `colSnapshot` instead of hardcoding column names